packages/flutter_test/lib/src/controller.dart

```diff
@@ -56,6 +56,18 @@ class SemanticsController {
 
   final WidgetsBinding _binding;
 
+  void _traverse(SemanticsNode node, bool Function(SemanticsNode node) onNode) {
+    final bool complete = onNode(node);
+    if (complete) {
+      return;
+    }
+
+    final List<SemanticsNode> children = node.debugListChildrenInOrder(DebugSemanticsDumpOrder.traversalOrder);
+    for(final SemanticsNode child in children) {
+      _traverse(child, onNode);
```

Comment:
if `onNode` returns true for `child`, do we still need to continue the for loop and _traverse the next child? I guess, I am a little confused what the semantics of returning true from onNode are...

Reply:
You are correct. Returning `true` from `onNode` will short circuit the traversal and prevent any further traversals.

Although, now that you mention it, it may continue traversing the `children` in already started loops. I'll probably have to pass the boolean in the actual `_traverse` method as well... Good catch! I'll update here when that's fixed.

Reply:
Should be fixed now!